plugins/github/api.ml

```diff
@@ -522,6 +568,102 @@ let head_of t repo id =
     | Some x -> Ok x
     | None -> Error (`Msg (Fmt.str "No such ref %a/%a" Repo_id.pp repo Ref.pp id))
 
+module CheckRun = struct
+  module Set_status = struct
+    let id = "github-check-run-set-status"
+
+    type nonrec t = t
+
+    module Key = struct
+      type t = {
+          commit : Commit_id.t;
+          context : string; (** TODO Make this app_id? *)
```

Comment:
@talex5 Wasn't sure if I could just make this `app_id` I don't seem to use `context` for anything useful?

Reply:
```suggestion
          check_name : string;
```
The appid should probably go in `Set_status.t` instead, assuming it's just used for access control.